src/Shared/Components/SelectPicker/SelectPickerTextArea.component.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useRef, useState } from 'react'
+import { InputActionMeta, SelectInstance, SingleValue } from 'react-select'
+
+import SelectPicker from './SelectPicker.component'
+import { SelectPickerOptionType, SelectPickerTextAreaProps } from './type'
+
+export const SelectPickerTextArea = ({
+    value,
+    options,
+    isCreatable,
+    onChange,
+    ...props
+}: SelectPickerTextAreaProps) => {
+    // STATES
+    const [inputValue, setInputValue] = useState((value as SingleValue<SelectPickerOptionType<string>>)?.value || '')
+
+    // REFS
+    const selectRef = useRef<SelectInstance<SelectPickerOptionType<string>>>(null)
+
+    useEffect(() => {
+        const selectValue = value as SingleValue<SelectPickerOptionType<string>>
+        setInputValue(selectValue?.value || '')
+    }, [value])
+
+    // METHODS
+    const onInputChange = (newValue: string, { action }: InputActionMeta) => {
+        if (action === 'input-change') {
+            setInputValue(newValue)
+            if (!newValue) {
+                onChange?.(null, {
+                    action: 'remove-value',
+                    removedValue: value as SingleValue<SelectPickerOptionType<string>>,
+                })
+            }
```

Comment:
we have a remove value as well in action meta, right?

Reply:
yes this is type-safe, remove-value is coming from action meta